### PR TITLE
python3Packages.airtouch4pyapi: make compatible with Home Assistant

### PR DESCRIPTION
**Pull request by dotlambda**
https://github.com/home-assistant/core/pull/71319 is not making progress

###### Description of changes
The wheel on PyPI differs from the source on GitHub, let's make them the same by using a commit from https://github.com/LonePurpleWolf/airtouch4pyapi/pull/10.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/airtouch4pyapi/default.nix [hunk 1]
```diff
@@ -16,8 +16,8 @@ buildPythonPackage rec {
     owner = "LonePurpleWolf";
     repo = pname;
     # https://github.com/LonePurpleWolf/airtouch4pyapi/issues/5
-    rev = "34783888846783c058fe79cec16feda45504f701";
-    sha256 = "17c7fm72p085pg9msvsfdggbskvm12a6jlb5bw1cndrqsqcrxywx";
+    rev = "5b5d91fad63495c83422e7a850897946ac95b25d";
+    hash = "sha256-lhKFsXbyEgx8+0h37kunV3IYZwcHGJCyFElzgxH/iZE=";
```
Comment (mweinelt) [inline pkgs/development/python-modules/airtouch4pyapi/default.nix:20]: Can we apply this as a patch? Making the change this way is quite intransparent.

Comment (dotlambda) [reply pkgs/development/python-modules/airtouch4pyapi/default.nix:20]: We can, but is that really so much better given that we're already fetching a random commit (which happens to be HEAD right now).

Comment (mweinelt) [reply pkgs/development/python-modules/airtouch4pyapi/default.nix]: This make src the upstream version and the patch indicates something is not alright upstream, which is … a statement.

Comment (dotlambda) [reply pkgs/development/python-modules/airtouch4pyapi/default.nix]: But it *is* upstream, just only on PyPI.

Comment (mweinelt) [reply pkgs/development/python-modules/airtouch4pyapi/default.nix]: Yeah, very weird all things considered.

**Pull request merged**